miasm2/core/asmblock.py

```diff
@@ -891,18 +1037,18 @@ def apply_splitting(self, symbol_pool, dis_block_callback=None, **kwargs):
 
     def __str__(self):
         out = []
-        for node in self.nodes():
-            out.append(str(node))
-        for nodeA, nodeB in self.edges():
-            out.append("%s -> %s" % (nodeA.label, nodeB.label))
+        for block in self.blocks:
+            out.append(str(block))
+        for label_a, label_b in self.edges():
+            out.append("%s -> %s" % (label_a, label_b))
         return '\n'.join(out)
 
     def __repr__(self):
         return "<%s %s>" % (self.__class__.__name__, hex(id(self)))
 
 # Out of _merge_blocks to be computed only once
-_acceptable_block = lambda block: (not isinstance(block, AsmBlockBad) and
-                                   len(block.lines) > 0)
+_acceptable_block = lambda graph, label: (not isinstance(graph.loc_key_to_block(label), AsmBlockBad) and
```

Comment:
If the graph contains `lockey` without block, this code will crash; it have to be also compared to `None` (like `AsmBlockBad`)

Reply:
done